Automattic/wp-calypso
# Extract user suggestions into top-level reducer

This PR extracts user suggestions into a `userSuggestions` top-level reducer, and modularises this new portion of state.

This removes the remainder of the unmodularised, top-level `users` portion of state.

#### Changes proposed in this Pull Request

* Extract `state.users.suggestions` into `state.userSuggestions`
* Modularise it

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/home` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and note that there's no `userSuggestions` key, even if you expand the list of keys. This indicates that the user suggestions state hasn't been loaded yet.
* Click `Reader` on the top bar.
* Verify that a `userSuggestions` key is added with the user suggestions state. This indicates that the user suggestions state has now been loaded.
* Smoke test user suggestion functionality in post comments (using `@`)

Follow-up to #48041. Fixes #42491.
